src/test/scala/nl/tradecloud/kafka/KafkaSubscriberSpec.scala

```diff
@@ -105,6 +106,68 @@ class KafkaSubscriberSpec extends TestKit(ActorSystem("KafkaSubscriberSpec", Con
 
       receiverProbe.expectNoMessage(defaultNegativeTimeout)
     }
+
+    "drop messages with an invalid type" in {
+      val receiverProbe = TestProbe()
+      val publisher = new KafkaPublisher(system)
+
+      val subscriber = new KafkaSubscriber(
+        serviceName = "test",
+        group = "test_group_0",
+        topics = Set("test_topic_5"),
+        system = system,
+        configurationProperties = Seq(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG -> "earliest")
+      )
+
+      subscriber.atLeastOnce[String](
+        Flow[KafkaMessage[String]].map { msg =>
+          receiverProbe.ref ! msg.msg
+
+          msg.offset
+        }
+      )
+
+      publisher.publish(
+        topic = "test_topic_5",
+        msg = Done
+      )
+      receiverProbe.expectNoMessage(defaultNegativeTimeout)
+    }
+
+    "commit messages using the commitFlow" in {
+      val receiverProbe = TestProbe()
+      val publisher = new KafkaPublisher(system)
+      val subscriber = new KafkaSubscriber(
+        serviceName = "test",
+        group = "test_group_0",
+        topics = Set("test_topic_6"),
+        system = system,
+        configurationProperties = Seq(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG -> "earliest")
+      )
+
+      var committedCount = 0
+      val control = subscriber
+        .consumerStream(
+          Flow[KafkaMessage[String]].map { msg =>
+            receiverProbe.ref ! msg.msg
+
+            msg.offset
+          }
+        )
+        .map { _ =>
+          committedCount += 1
```

Comment:
Could you please explain me (by Zoom) how this tests the commit of the divertTo flow

Reply:
The consumerStream in the KafkaSubscriber contains the Source and all flow shapes which are run when a Sink is connected. The last flow shape in the consumerStream is the commitFlow, committing is performed in this flow. 

The test adds a map flow and a Sink to this stream and if an element reaches this map flow, it must have gone through the commitFlow. Thereby proving the commitFlow is passed by each element, which meets the consumerStream conditions (type check etc).